src/main/java/edu/umich/ctools/sectionsUtilityTool/FriendServlet.java

```diff
@@ -17,235 +17,353 @@
+        String fullSocialLoginApiUrl = url + "/guest/invite";
+
+        try {
+            response.setContentType("application/json");
+            request.setCharacterEncoding(Utils.CONSTANT_UTF_8);
+            StringEntity apiPostBodyStringEntity;
+            out = response.getWriter();
+            Utils.logApiCall(request);
+
+            String nonUmichEmail = URLEncoder.encode(request.getParameter(PARAMETER_ID), Utils.CONSTANT_UTF_8).replace("%40", "@");
+            String nonUmichEmailFirstName = URLEncoder.encode(request.getParameter(PARAMETER_FRIEND_FIRST_NAME), Utils.CONSTANT_UTF_8);
+            String nonUmichEmailLastName = URLEncoder.encode(request.getParameter(PARAMETER_FRIEND_LAST_NAME), Utils.CONSTANT_UTF_8);
+            String keySecretStr = String.format("%s:%s", key, secret);
+
+            String encoding = Base64.getEncoder().encodeToString((keySecretStr).getBytes());
+            CloseableHttpClient httpclient = HttpClients.createDefault();
+            HttpPost httpPost = new HttpPost(fullSocialLoginApiUrl);
+            httpPost.setHeader("Accept", Utils.APPLICATION_JSON);
+            httpPost.setHeader("Content-type", Utils.APPLICATION_JSON);
+            httpPost.setHeader(HttpHeaders.AUTHORIZATION, "Basic " + encoding);
+            apiPostBodyStringEntity = socialLoginApiRequestBody(spURL, nonUmichEmail, nonUmichEmailFirstName, nonUmichEmailLastName);
+            httpPost.setEntity(apiPostBodyStringEntity);
+            HttpResponse res = httpclient.execute(httpPost);
+            if (res == null) {
+                response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+                json = buildResponseObject("error", "Non-Umich Account NOT created for due to No response", fullSocialLoginApiUrl);
+                out.print(json.toString());
+                out.flush();
+                return;
+            }
+
+            int statusCode = res.getStatusLine().getStatusCode();
+            String apiResponse = EntityUtils.toString(res.getEntity(), Utils.CONSTANT_UTF_8);
+            response.setStatus(statusCode);
+            M_log.info(String.format("Social Login API Response code %s and details %s ", statusCode, apiResponse));
+            if (statusCode == HttpStatus.SC_CREATED) {
+                String detailedMessage = String.format("Non-Umich Account created for %s ", nonUmichEmail);
+                buildResponseObject("created", detailedMessage, fullSocialLoginApiUrl);
+                json = buildResponseObject("created", detailedMessage, fullSocialLoginApiUrl);
+            } else {
+                json = buildResponseObject("error", apiResponse, fullSocialLoginApiUrl);
+            }
+        } catch (IOException e) {
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            json = buildResponseObject("error", "Non-Umich Account NOT created for due to " + e.getMessage(), fullSocialLoginApiUrl);
```

Comment:
same here: need to log more details for the parameter

Reply:
I have fix for this here https://github.com/tl-its-umich-edu/canvas_course_manager/pull/266/commits/484e2800e5f801745ae1f98df6c19d06cd8e90d8#r346471959